--- a/finalforeach/cosmicreach/networking/packets/blocks/PlaceBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/blocks/PlaceBlockPacket.java
@@ -15,6 +_,7 @@
 import finalforeach.cosmicreach.world.Zone;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
+import sh.miles.voidcr.impl.world.block.VoidBlockState;
 
 public class PlaceBlockPacket extends GamePacket {
    BlockPosition blockPos;
@@ -65,6 +_,17 @@
                         itemSlot.setDirty(true);
                         player.inventory.syncDirtySlots(0, identity);
                      } else {
+                        // VoidCR - start PrePlayerPlaceBlockEvent
+                        if (identity.getZone() != zone) return; // return if the player tries to place a block in another zone
+                        final var event = sh.miles.voidcr.impl.plugin.lifecycle.event.VoidEventFactory.prePlayerPlaceBlock(identity, this.blockPos, this.targetBlockState);
+                        if (event != null && event.isCanceled()) {
+                           // re-sync world and inventory
+                           itemSlot.setDirty(false);
+                           player.inventory.syncDirtySlots(0, identity);
+                           identity.send(new finalforeach.cosmicreach.networking.packets.blocks.BlockReplacePacket(zone, blockPos.getBlockState(), this.blockPos));
+                        }
+                        if (event != null) this.targetBlockState = ((VoidBlockState)event.getState()).getMirror();
+                        // VoidCR - end PrePlayerPlaceBlockEvent
                         BlockEventTrigger[] triggers = this.targetBlockState.getTrigger("onPlace");
                         if (triggers != null) {
                            BlockEventArgs args = new BlockEventArgs();
@@ -88,6 +_,7 @@
                }
             }
          }
+          sh.miles.voidcr.impl.plugin.lifecycle.event.VoidEventFactory.postPlayerPlaceBlock(identity, this.blockPos, this.targetBlockState); // VoidCR - PostPlayerPlaceBlockEvent
       }
    }
 }
