--- a/finalforeach/cosmicreach/entities/player/PlayerEntity.java
+++ b/finalforeach/cosmicreach/entities/player/PlayerEntity.java
@@ -192,7 +_,7 @@
 
    @Override
    public void onAttackInteraction(Entity sourceEntity) {
-      if (sourceEntity instanceof PlayerEntity) {
+       if (sourceEntity instanceof PlayerEntity) {
          if (!ServerSettings.PVP_ENABLED.isFalse()) {
             this.hit(sourceEntity, 1.0F);
          }
