--- a/finalforeach/cosmicreach/networking/netty/NettyServer.java
+++ b/finalforeach/cosmicreach/networking/netty/NettyServer.java
@@ -89,6 +_,7 @@
                .option(ChannelOption.SO_BACKLOG, 128))
             .childOption(ChannelOption.SO_KEEPALIVE, true);
          this.channelFuture = b.bind(this.port).sync();
+         /* VoidCR start - smart console
          String inputString = null;
          Logger.info("Server started, type 'stop' to shutdown and 'help' for commands.");
 
@@ -101,6 +_,13 @@
             }
          }
 
+          */
+         final var console = new sh.miles.voidcr.adjust.rewrite.ConsoleRewrite();
+         console.spin();
+         while(ServerSingletons.running.get());
+         console.halt();
+         sh.miles.voidcr.impl.server.VoidServer.SERVER.unload(); // VoidCR - unload server
+         // VoidCR end - smart console
          this.shutdown();
          this.channelFuture.channel().closeFuture().sync();
       } finally {
