--- a/finalforeach/cosmicreach/gameevents/blockevents/BlockEventTrigger.java
+++ b/finalforeach/cosmicreach/gameevents/blockevents/BlockEventTrigger.java
@@ -18,7 +_,7 @@
          throw new RuntimeException("Could not find action for id: " + actionId);
       } else {
          try {
-            this.action = (IGameEventAction)json.readValue(actionClass, jsonData);
+            this.action = (IBlockAction) json.readValue(actionClass, jsonData); // VoidCR decompile fix
             this.condition = null;
             JsonValue rootCondition = jsonData.get("if");
             if (rootCondition != null) {
