--- a/finalforeach/cosmicreach/blocks/blockentities/BlockEntityLaserEmitter.java
+++ b/finalforeach/cosmicreach/blocks/blockentities/BlockEntityLaserEmitter.java
@@ -15,6 +_,12 @@
 import finalforeach.cosmicreach.world.Zone;
 
 public class BlockEntityLaserEmitter extends CustomUIBlockEntity implements IBlockEntity {
+   // VoidCR add fetcher
+   @Override
+   public sh.miles.voidcr.impl.world.block.entity.VoidBlockEntity<?> getVoidMirror() {
+      return cache == null ? cache = new sh.miles.voidcr.impl.world.block.entity.VoidBlockEntityLaserEmitter(this) : cache;
+   }
+   // VoidCR add fetcher
    public static final String BLOCK_ENTITY_ID = "base:laser_emitter";
    public static final String SIGNAL_SHOOT_PROJECTILE = "shootProjectile";
    @CRBSerialized
