--- a/finalforeach/cosmicreach/blocks/blockentities/BlockEntity.java
+++ b/finalforeach/cosmicreach/blocks/blockentities/BlockEntity.java
@@ -15,6 +_,10 @@
 import java.util.HashMap;
 
 public abstract class BlockEntity implements IBlockEntity, ICRBinSerializable, ISlotContainerParent {
+   // VoidCR - apply generic API getter
+   protected transient sh.miles.voidcr.impl.world.block.entity.VoidBlockEntity<?> cache;
+   public abstract sh.miles.voidcr.impl.world.block.entity.VoidBlockEntity<?> getVoidMirror();
+   // VoidCR - apply generic API getter
    public transient HashMap<String, Runnable> signalMap;
    Zone zone;
    int x;
