--- a/finalforeach/cosmicreach/blocks/blockentities/BlockEntityDispenser.java
+++ b/finalforeach/cosmicreach/blocks/blockentities/BlockEntityDispenser.java
@@ -9,6 +_,12 @@
 import finalforeach.cosmicreach.world.Zone;
 
 public class BlockEntityDispenser extends CustomUIBlockEntity {
+    // VoidCR add fetcher
+    @Override
+    public sh.miles.voidcr.impl.world.block.entity.VoidBlockEntity<?> getVoidMirror() {
+        return cache == null ? cache = new sh.miles.voidcr.impl.world.block.entity.VoidBlockEntityDispenser(this) : cache;
+    }
+    // VoidCR add fetcher
    private static final String SHOOT_DISPENSER_SIGNAL = "shootDispenser";
    public static final String BLOCK_ENTITY_ID = "base:dispenser";
    public boolean shootPending = false;
