--- a/finalforeach/cosmicreach/blocks/blockentities/BlockEntityWorkbench.java
+++ b/finalforeach/cosmicreach/blocks/blockentities/BlockEntityWorkbench.java
@@ -5,6 +_,12 @@
 import finalforeach.cosmicreach.world.Zone;
 
 public class BlockEntityWorkbench extends CustomUIBlockEntity {
+    // VoidCR add fetcher
+    @Override
+    public sh.miles.voidcr.impl.world.block.entity.VoidBlockEntity<?> getVoidMirror() {
+        return cache == null ? cache = new sh.miles.voidcr.impl.world.block.entity.VoidBlockEntityWorkbench(this) : cache;
+    }
+    // VoidCR add fetcher
    public static final String BLOCK_ENTITY_ID = "base:workbench";
 
    public BlockEntityWorkbench(
