--- a/finalforeach/cosmicreach/server/ServerLauncher.java
+++ b/finalforeach/cosmicreach/server/ServerLauncher.java
@@ -33,6 +_,7 @@
             }
          }
 
+         sh.miles.voidcr.impl.server.VoidServer.SERVER.getPluginLoader().loadPlugins(); // VoidCR - load plugins
          ServerSingletons.create();
          String serverWorldName = "server-world-1";
          GameSingletons.world = ChunkLoader.loadWorld(serverWorldName);
