--- a/finalforeach/cosmicreach/ZoneLoaders.java
+++ b/finalforeach/cosmicreach/ZoneLoaders.java
@@ -60,7 +_,7 @@
             }
          }
 
-         if (GameSingletons.getNumberOfPlayers() == 0 && !GameSingletons.isClient || this.zoneLoaders.isEmpty()) {
+         if (GameSingletons.getNumberOfPlayers() == 0 && finalforeach.cosmicreach.networking.server.ServerSingletons.running.get() && !GameSingletons.isClient || this.zoneLoaders.isEmpty()) {
             synchronized (worldGenThread) {
                if (GameSingletons.getNumberOfPlayers() == 0 || this.zoneLoaders.isEmpty()) {
                   Iterator i$x = this.zoneLoaders.entries().iterator();
