--- a/finalforeach/cosmicreach/singletons/GameSingletonPlayers.java
+++ b/finalforeach/cosmicreach/singletons/GameSingletonPlayers.java
@@ -41,6 +_,7 @@
          World world = GameSingletons.world;
          if (world != null) {
             world.players.clear();
+            world.nameToPlayer.clear();
 
             for (Zone zone : world.getZones()) {
                zone.getPlayers().clear();
