--- a/finalforeach/cosmicreach/singletons/GameSingletons.java
+++ b/finalforeach/cosmicreach/singletons/GameSingletons.java
@@ -212,7 +_,7 @@
    public static <T extends CustomScreenInfoRemoteControl> void registerCustomScreenOpener(
       Class<T> clazz, Consumer<T> info
    ) {
-      infoScreenOpeners.put(clazz, info);
+      infoScreenOpeners.put(clazz, (Consumer<CustomScreenInfoRemoteControl>) info); // VoidCR - decompile fix
    }
 
    public static boolean isDoneLoadingGameObjects() {
